modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/ManagementToolbarTag.java

```diff
@@ -1102,7 +1102,7 @@ protected int processStartTag() throws Exception {
 			LinkTag linkTag = new LinkTag();
 
 			linkTag.setCssClass("component-link tbar-link");
-			linkTag.setHref(getClearResultsURL());
+			linkTag.setHref(HtmlUtil.escapeAttribute(getClearResultsURL()));
```

Comment:
Hey @julien 

There are some occurrences in the portal (for instance: https://github.com/liferay/liferay-portal-ee/blob/263ea30ad1e562a9b671ae4f9f70021c074a8760/modules/apps/configuration-admin/configuration-admin-web/src/main/resources/META-INF/resources/search_results.jsp#L20) where we are already escaping the attribute, Doesn't this may cause a double escaping?

Reply:
Hey @victorg1991 thanks for the heads up. I can check that, do you know how I test this correctly?

Reply:
Doesn't seem like it's the case @victorg1991

Reply:
I had to make a small change to actually see it (without the JS component), but I can see that it does cause double escaping 

For example, when searching something in system settings, this is what the `clearResultsURL` looks like

```
http://localhost:8080/group/control_panel/manage?p_p_id=com_liferay_configuration_admin_web_portlet_SystemSettingsPortlet&amp;p_p_lifecycle=0&amp;p_p_state=maximized&amp;p_p_mode=view
```

Maybe the easier option would be to **not** render those URLs in the JSP, but I'm not sure that's a perfect solution.

Maybe our Java experts (@nhpatt or @javierdearcos) have a better idea?

Reply:
Using a disabled button seems like a good idea, I'll probably also avoid rendering the `searchActionURL` in the JSP to avoid the other XSS issue

Reply:
So with the change in https://github.com/liferay-frontend/liferay-portal/pull/1666/commits/1605e8788db30af30aaf3dd88cefac27045433dd, until the JS component is loaded it will look like this 
![](https://user-images.githubusercontent.com/5572/141795570-cf93423b-3676-48d8-83e6-e5be9408ae1a.png)
and once the JS has loaded, it will look just like it used to 
![](https://user-images.githubusercontent.com/5572/141795633-1db0f926-a44d-409d-8c98-0bd046d96f7b.png)

Reply:
We may also want to disable the input while the component is loading, or it is not needed? I guess the user should be super fast to click on the input, add some words and press enter

Reply:
I don't think it's needed